server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -293,6 +292,7 @@ public static SearchHit unpooled(int docId, String id) {
     }
 
     public static SearchHit unpooled(int nestedTopDocId, String id, NestedIdentity nestedIdentity) {
+        // always referenced search hits do NOT call #deallocate
```

Comment:
This might seem as noise but it helped me when making sense of the pool/unpool stuff so thought maybe it's useful to future spelunkers.